lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -104,12 +103,12 @@
     private final List<SessionPropertyAccessControlRule> sessionPropertyRules;
     private final Set<AnySchemaPermissionsRule> anySchemaPermissionsRules;
 
-    public FileBasedAccessControl(CatalogName catalogName, File configFile)
+    public FileBasedAccessControl(CatalogName catalogName, Supplier<AccessControlRules> rulesProvider)
     {
         this.catalogName = catalogName.toString();
 
-        AccessControlRules rules = parseJson(configFile.toPath(), AccessControlRules.class);
-        checkArgument(!rules.hasRoleRules(), "File connector access control does not support role rules: %s", configFile);
+        AccessControlRules rules = rulesProvider.get();
```

Comment:
requireNonNull check for supplier ? Any specific reason for supplier ? Can we inject AccessControlRules directly ?